hazelcast/serialization/portable/reader.py

```diff
@@ -406,6 +520,41 @@ def read_string(self, field_name):
         self.validate_type_compatibility(fd, FieldType.STRING)
         return super(MorphingPortableReader, self).read_string(field_name)
 
+    def read_decimal(self, field_name):
+        fd = self._class_def.get_field(field_name)
+        if fd is None:
+            return None
+        self.validate_type_compatibility(fd, FieldType.DECIMAL)
+        return super(MorphingPortableReader, self).read_decimal(field_name)
```

Comment:
this is duplicated a lot in this reader. can you extract it to a function and re-use it?

Reply:
Introduced new `_validate_type_compatibility_and_read` function to prevent duplicate at https://github.com/hazelcast/hazelcast-python-client/pull/569/commits/9aff832066d0c194a9e3cb27d8a53950d629388a.